app/models/order_item.rb

```diff
@@ -36,12 +50,13 @@ def item_price # v. price / cost / item_cost / total_item_price / total_item_cos
 
   def product_has_stock? # !Q is this the right way to do this?
     # stock = product.has_available_stock?
-    unless product.has_available_stock?
+    if product.has_available_stock?
```

Comment:
Does this method return true/false? I think this could be refactored so that the result of .has_available_stock? is returned, and the error message set elsewhere?